### DISCUSSION TITLE
Enabling auto-merge (and some extra branch protection rules)

### DISCUSSION THREAD

**Discussion by DavidBiddle** [govuk-forms/forms#44] [Ideas]
I'd like to propose that we enable auto-merging on our pull requests - this would mean that at any point after creating a PR you could set it to merge automatically when all checks pass. This would marginally reduce manual effort, and reduce the risk of approved and ready PRs sitting around without being merged, and eventually going stale. 

As a counterbalance to this, I'd also want to enable the following branch protection rules on `main` for each repo:
- Require conversation resolution before merging
- Require status checks to pass before merging
  - Require branches to be up to date before merging

This would make sure that only PRs which are actually ready get merged.

**Comment by 0atman** [comment]
I'm interested in this. Would minimum numbers of code reviewers also be a prerequisite?

**Reply by aliuk2012** [reply]
At the moment there is already a minimum of 1 review before pr can be merged.

**Reply by DavidBiddle** [reply]
Think we already have a minimum of 1 enabled. We could set it higher if we feel the need

**Reply by DavidBiddle** [reply]
Oops, bad timing!

**Reply by aliuk2012** [reply]
great timing....i'd say ;)

**Reply by aliuk2012** [reply]
We could bump this to min. of 2 now that we are a bigger team.

**Comment by aliuk2012** [comment]
With auto-merge and CD, a ticket could be done as soon as someone approves it.

**Reply by thomasiles** [reply]
Would this feature make things harder on the PR owner if they now have responsibility for deploying merged code (in a timely manner)?

**Reply by DavidBiddle** [reply]
Not any more 😎
